webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/reports/execution/ReportsExecutionService.scala

```diff
@@ -195,18 +193,19 @@ class ReportsExecutionService (
     Future {
       //update changes by rules
       (for {
-        changes <- reportsRepository.getChangeReportsOnInterval(lowestId, highestId)
-        updated <- cachedChanges.update(changes)
+        changes         <- reportsRepository.getChangeReportsOnInterval(lowestId, highestId)
```

Comment:
as an afterthought, it should probably be the getChangeReportsOnInterval which ought to have a different query

Reply:
Afterthought of #15041?

Reply:
OK, after some more hours on the code, I understand that you want to change the query to only return the relevant part (ruleId, executionTimestamp). That seems to be a much better solution to minimize memory, I agree!